consensus/types/build.rs

```diff
@@ -0,0 +1,22 @@
+use std::env;
+use std::io::Write;
+use std::path::Path;
+
+fn main() {
+    let mut file =
+        std::fs::File::create(&Path::new(&env::var("OUT_DIR").unwrap()).join("subnet_id_map.rs"))
+            .expect("creating source file failed");
+    file.write_all(b"pub fn int_to_str(i: u64) -> &'static str { match i {")
```

Comment:
I appreciate the rawness of this, but perhaps we could do the same thing with `lazy_static` instead of generating this code here.

I was thinking we could initialize a `ToStringCache` in a `lazy_static!`. It would have a one-time runtime cost, but I think that's totally fine. The benefit is that we stay inside the type system and I'd say it's easier to read than code that generates code (that isn't a macro).

Reply:
yes maybe code generation was a little bit over the top here ;), I updated the PR now. The "problem" that we have to hard code the maximum subnet id remains.